File: Proj/User-UI/src/app/Service/User.service.ts
import { Injectable } from '@angular/core';
import { CustomHttpService } from './CustomHttp.service';
import { Observable } from 'rxjs/Rx';
import { CommonUtil } from '../common/commonutil';
import { DataTableParam } from '../Model/DataTableParam';

@Injectable()
export class UserService {

    constructor(private customHttpService: CustomHttpService) { }

    saveUser(object: any): Observable<any> {
        const url = CommonUtil.getApiEndPoints() + '/user/add';
        const body = JSON.stringify({valueObject: object});
        return this.customHttpService.post(url, body);
    }

    listUser(dataTableParam: DataTableParam): Observable<any> {
        const url = CommonUtil.getApiEndPoints() + '/user/list';
        const body = JSON.stringify({'dataTableParam': dataTableParam});
        return this.customHttpService.post(url, body);
    }

    loadUserById(id: any): Observable<any> {
        const url = CommonUtil.getApiEndPoints() + '/user/loadUserById';
        const body = JSON.stringify({valueObject: id});
        return this.customHttpService.post(url, body);
    }

    deleteUser(id: any): Observable<any> {
        const url = CommonUtil.getApiEndPoints() + '/user/delete';
        const body = JSON.stringify({valueObject: id});
        return this.customHttpService.post(url, body);
    }

}

File: Proj/server/demoMysql/src/main/java/com/mysql/demoMysql/service/UserService.java
package com.mysql.demoMysql.service;


import com.mysql.demoMysql.model.DataTableEntity;
import com.mysql.demoMysql.model.DataTableParam;
import com.mysql.demoMysql.model.User;

public interface UserService {

	User saveUser(User user) throws Exception;
	
	DataTableEntity<User> getUserList(DataTableParam datatableParam) throws Exception;
	
	User getUserById(String id) throws Exception;
	
	void deleteUserById(String id) throws Exception;
	
}
